### DISCUSSION TITLE
export 'useId' (imported as 'e') was not found in 'vue'

### DISCUSSION THREAD

**Discussion by AZCodingAccount** [tailwindlabs/headlessui#3474] [Help] [answered]
Version: 
`vue: 3.4.31`
`@headlessui/vue@1.7.23_vue@3.4.31_typescript@5.6.2_`

Hello, I encountered problems during `pnpm build`, but running `pnpm dev` works normally, which is strange. Could it be a version issue? However, I see that `headlessui` has the following configuration in peer dependencies:
```json
"peerDependencies": {
  "vue": "^3.2.0"
},
```

screenshots：
![image](https://github.com/user-attachments/assets/4a0d61ba-203e-4ef8-b91f-08485cac5b56)

**Comment by kaz1ngton** [comment]
@AZCodingAccount did you find solution?

**Reply by AZCodingAccount** [reply, marked as answer]
I locked the version of headless-ui, which solved this issue. You can add this configuration item in the package.json to prevent it from updating to the latest version:

```json
"pnpm": {
  "overrides": {
    "@headlessui/vue": "1.7.20"
  }
}
